unlock-app/src/components/interface/locks/List/elements/LockCard.tsx

```diff
@@ -140,18 +175,55 @@ const LockIcon = ({ lock }: LockIconProps) => {
   )
 }
 
-export const LockCard = ({ lock, network, isLoading }: LockCardProps) => {
+export const LockCard = ({ lock, network }: LockCardProps) => {
   const { networks } = useConfig()
-  const lockAddress = lock.address
+  const web3service = useWeb3Service()
+  const tokenAddress = lock?.tokenAddress
+  const lockAddress = lock?.address
   const [isCopied, setCopied] = useClipboard(lockAddress, {
     successDuration: 2000,
   })
-
   const { explorer, baseCurrencySymbol } = networks?.[network] ?? {}
 
   const explorerUrl = explorer?.urls?.address(lockAddress) || '#'
 
-  const symbol = (lock as any)?.currencySymbol ?? baseCurrencySymbol
+  const getBalance = async (
+    address: string,
+    chainId: number,
+    tokenAddress: string
+  ) => {
+    return await web3service.getAddressBalance(
+      address,
+      chainId,
+      tokenAddress === DEFAULT_USER_ACCOUNT_ADDRESS ? undefined : tokenAddress
+    )
+  }
+
+  const getSymbol = async () => {
```

Comment:
I wonder if things could be more elegant if combined in a single function that executes both promises?